packages/material-ui/src/SwipeableDrawer/SwipeArea.js

```diff
@@ -1,50 +1,77 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import withStyles from '../styles/withStyles';
+import {
+  unstable_composeClasses as composeClasses,
+  generateUtilityClass,
+} from '@material-ui/unstyled';
+import experimentalStyled from '../styles/experimentalStyled';
 import capitalize from '../utils/capitalize';
 import { isHorizontal } from '../Drawer/Drawer';
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
-    position: 'fixed',
-    top: 0,
-    left: 0,
-    bottom: 0,
-    zIndex: theme.zIndex.drawer - 1,
+export function getSwipeAreaUtilityClass(slot) {
+  return generateUtilityClass('MuiSwipeableDrawer', slot);
+}
+
+const SwipeAreaRoot = experimentalStyled(
+  'span',
+  {},
+  {
+    name: 'PrivateSwipeArea',
+    slot: 'Root',
```

Comment:
I think that we should remove this as well as the global class names. This component is private. Same as https://github.com/mui-org/material-ui/blob/77e4e292e3efa46640c02e92c91beb76f1099bbf/packages/material-ui/src/OutlinedInput/NotchedOutline.js#L7
?

Reply:
Fair enough

Reply:
If we need to customize it, I think that SwipeableDrawer could create a new slot for it and apply it as a class name. We have used this approach in the past.

For the pickers, we went in a different direction with @dmtrKovalenko. We made the classes API of the internal component's public to ease the customization. It's an oxymoron but I'm not sure we have better alternatives.

Reply:
The reason why I added the utility classes is because the `SwipeableDrawer` is not styled at all. That’s why I thought I could add this as root and make it available for customization. I would even add the overrides resolver here, as there is no other point for injecting customization.

Reply:
I forgot that SwipeableDrawer had no styles. If we want to make SwipeArea public in the future, I think that the approach used here will make sense.